### PR TITLE
Adding the response to the request module

### PR DESCRIPTION
**Pull request by melyxlin**
This PR is an extension of #25, in which completes the functionality of the request module. It enables the web app to display the response from the HTTP request using the inputted token.

### CODE REVIEW

File: web/client/src/util/apiWrapper.js [hunk 1]
```diff
@@ -19,3 +19,28 @@ export const validateToken = (requestOptions) => {
...
+  for (i = 0; i < values.headers.length; i++) {
+    if (
+      values.headers[i].headerName.length !== 0 &&
+      values[i].headerValue.length !== 0
+    ) {
+      param.headers[values.headers[i].headerName] =
+        values.headers[i].headerValue;
+    }
+  }
+  return axios(param).catch(function (error) {
```
Comment (yousefa00) [inline web/client/src/util/apiWrapper.js:43]: Use arrow function here

File: web/client/src/components/request/RequestModule.jsx [hunk 2]
```diff
@@ -17,18 +17,38 @@ import {
...
 export default function RequestModule() {
   const [openHeaderBox, setOpenHeaderBox] = useState(false);
   const [openRequestBodyBox, setOpenRequestBodyBox] = useState(false);
+  const [response, setResponse] = useState("");
+  const [errorOpen, setErrorOpen] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [errMessage, setErrMessage] = useState("");
+
+  const getResponse = async (values) => {
+    const Response = await getHTTPResponse(values);
```
Comment (yousefa00) [inline web/client/src/components/request/RequestModule.jsx:43]: Lower case var

File: web/client/src/components/request/RequestModule.jsx [hunk 3]
```diff
@@ -256,12 +273,38 @@ export default function RequestModule() {
                 multiline
                 fullWidth
                 variant="outlined"
-                value={JSON.stringify(values, null, 2)}
+                // value={JSON.stringify(values, null, 2)}
```
Comment (yousefa00) [inline web/client/src/components/request/RequestModule.jsx:276]: Remove comment if not needed anymore

File: web/client/src/util/apiWrapper.js [hunk 4]
```diff
@@ -19,3 +19,28 @@ export const validateToken = (requestOptions) => {
     return JSON.parse(JSON.stringify({ Error: error }));
   });
 };
+
+export const getHTTPResponse = (values) => {
+  const param = {
+    url: values.URI,
```
Comment (yousefa00) [inline web/client/src/util/apiWrapper.js:25]: URI => URL?

File: web/client/src/util/apiWrapper.js [hunk 5]
```diff
@@ -19,3 +19,28 @@ export const validateToken = (requestOptions) => {
...
+    method: values.httpMethod.toLowerCase(),
+    headers: {
+      Authorization: "Bearer " + values.token,
+      ...(values.contentType ? { "Content-Type": values.contentType } : {}),
+      ...(values.reqBody ? { data: values.reqBody } : {}),
+    },
+  };
+  let i;
+  for (i = 0; i < values.headers.length; i++) {
```
Comment (yousefa00) [inline web/client/src/util/apiWrapper.js:34]: Move the `let i` statement into the loop definition: `for (let i = 0; i < values.headers.length; i++) { ... }`

File: web/client/src/util/apiWrapper.js [hunk 6]
```diff
@@ -19,3 +19,28 @@ export const validateToken = (requestOptions) => {
...
+    if (
+      values.headers[i].headerName.length !== 0 &&
+      values[i].headerValue.length !== 0
+    ) {
+      param.headers[values.headers[i].headerName] =
+        values.headers[i].headerValue;
+    }
+  }
```
Comment (yousefa00) [inline web/client/src/util/apiWrapper.js:42]: This validation should probably be done in the request module and prevent a user from making the request until the conditions are met.

File: web/client/src/util/prettify_json.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        }
+      } else if (/true|false/.test(match)) {
+        cls = "boolean";
+      } else if (/null/.test(match)) {
+        cls = "null";
+      }
+      return [cls, match];
+    }
+  );
+};
```
Comment (yousefa00) [inline web/client/src/util/prettify_json.js:25]: Ensure proper license for where you got this code from.

File: web/client/src/util/apiWrapper.js (hunk 1 above)
Comment (melyxlin) [reply web/client/src/util/apiWrapper.js]: done!

File: web/client/src/components/request/RequestModule.jsx (hunk 2 above)
Comment (melyxlin) [reply web/client/src/components/request/RequestModule.jsx]: done! Fixed also for other files as well

File: web/client/src/components/request/RequestModule.jsx (hunk 3 above)
Comment (melyxlin) [reply web/client/src/components/request/RequestModule.jsx]: done!

File: web/client/src/util/apiWrapper.js (hunk 4 above)
Comment (melyxlin) [reply web/client/src/util/apiWrapper.js]: changed to url

File: web/client/src/util/apiWrapper.js (hunk 5 above)
Comment (melyxlin) [reply web/client/src/util/apiWrapper.js]: done!

File: web/client/src/util/apiWrapper.js (hunk 6 above)
Comment (melyxlin) [reply web/client/src/util/apiWrapper.js]: moved to request module

File: web/client/src/util/prettify_json.js (hunk 7 above)
Comment (melyxlin) [reply web/client/src/util/prettify_json.js]: done!

File: web/client/src/util/apiWrapper.js [hunk 8]
```diff
@@ -3,7 +3,7 @@ import axios from "axios";
 const BASE_URL =
   process.env.NODE_ENV === "production"
     ? "https://yousefa-step-2020.uc.r.appspot.com/api"
-    : "http://localhost:8080/api";
+    : "https://yousefa-step-2020.uc.r.appspot.com/api";
```
Comment (yousefa00) [inline web/client/src/util/apiWrapper.js:6]: Don't forget to change this back

Comment (melyxlin) [reply web/client/src/util/apiWrapper.js]: done!

**Pull request merged**